Grains return wrong OS version and other OS related values for Oracle Linux

galet opened the pull request:
OEL contains space characters at the end of some lines. The current regular expression doesn't expect that. This causes various issues especially for "service" module which doesn't work on OEL.

Fixes Issue #29425

Commit:
Grains return wrong OS version and other OS related values for Oracle Linux

OEL contains space characters at the end of some lines. The current regular expression doesn't expect that. This causes various issues especially for "service" module which doesn't work on OEL.

basepi commented:
Right here you're just matching the whitespace. Should we also strip it? Currently we only strip the newline at the end.

basepi commented:
Actually, I wonder if instead of changing the regex, we should just add the space character to the line.strip.....

galet commented:
You are right, that would be better.

Commit:
Grains return wrong OS version and other OS related values for Oracle Linux

Updating based on feedback - strip all whitespace characters instead of updating the regex

galet commented:
Actually the strip without any arguments will strip all whitespace characters which would be better IMHO.

jfindlay commented:
Thanks, @galet.

aabognah commented:
salt 'server-name' grains.item osfinger

returns:

    server-name:
    ----------
    osfinger:
        Oracle Linux Server"-6

Should the double quote be there?
salt 2015.8.3 on OEL6

aabognah commented:
same thing with osfullname:

        osfullname:
        Oracle Linux Server"
